File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        n = len(temperatures)
        res = [0] * n
        for i in range(n):
            t = temperatures[i]
            while stack and t > temperatures[stack[-1]]:
                j = stack.pop()
                res[j] = i - j
            stack.append(i)
        return res

File: Data Structures & Algorithms/simplify-path/submission-0.java
class Solution {
    public String simplifyPath(String path) {
        Deque<String> stack = new LinkedList<>();
        StringBuilder dir = new StringBuilder();
        path += '/';
        for (char c : path.toCharArray()) {
            if (c != '/') {
                dir.append(c);
            } else {
                String currDir = dir.toString();
                if (currDir.equals("..")) {
                    stack.pollLast();
                } else if (currDir.equals(".")) {
//                    Do nothing
                } else if (!currDir.isBlank()){
                    stack.add(currDir);
                }
                dir.setLength(0);
            }
        }


        StringBuilder res = new StringBuilder("/");
        for (String d : stack) {
            res.append(d)
                    .append("/");
        }
        if (res.length() > 1) {
            res.deleteCharAt(res.length() - 1);
        }
        return res.toString();
    }
}